File: packages/php/src/Resource/ViewerTokens.php
<?php

declare(strict_types=1);

namespace LogSeal\Resource;

use LogSeal\LogSeal;

class ViewerTokens
{
    public function __construct(private readonly LogSeal $client) {}

    /**
     * Create a short-lived viewer token for the embeddable log viewer.
     */
    public function create(string $organizationId, ?int $expiresIn = null): array
    {
        $body = ['organization_id' => $organizationId];
        if ($expiresIn !== null) $body['expires_in'] = $expiresIn;
        return $this->client->request('POST', '/v1/viewer-tokens', $body);
    }
}

File: packages/ruby/lib/logseal/resources/viewer_tokens.rb
# frozen_string_literal: true

module LogSeal
  module Resources
    class ViewerTokens
      def initialize(client)
        @client = client
      end

      # Create a short-lived viewer token for the embeddable log viewer.
      def create(organization_id:, expires_in: nil)
        body = { organization_id: }
        body[:expires_in] = expires_in if expires_in
        @client.request(:post, "/v1/viewer-tokens", body:)
      end
    end
  end
end
